File: go/0217.contains-duplicate/solution.go
// Created by Olgierd Palasz at 2026/05/06 21:34
// leetgo: dev
// https://leetcode.com/problems/contains-duplicate/

package main

import (
	"bufio"
	"fmt"
	"os"

	. "github.com/j178/leetgo/testutils/go"
)

// @lc code=begin

func containsDuplicate(nums []int) bool {
	Hash := make(map[int]bool)
	for _, i := range nums {
		if Hash[i] {
			return true
		} else {
			Hash[i] = true
		}
	}
	return false
}

// @lc code=end

func main() {
	stdin := bufio.NewReader(os.Stdin)
	nums := Deserialize[[]int](ReadLine(stdin))
	ans := containsDuplicate(nums)

	fmt.Println("\noutput:", Serialize(ans))
}

File: rust/src/0217.contains-duplicate/solution.rs
// Created by Olgierd Palasz at 2026/07/14 20:34
// leetgo: dev
// https://leetcode.com/problems/contains-duplicate/

use anyhow::Result;
use leetgo_rs::*;

struct Solution;

// @lc code=begin
use std::collections::HashSet;
impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut seen: HashSet<i32> = HashSet::with_capacity(nums.len());
        nums.into_iter().any(|n| !seen.insert(n))
    }
}

// @lc code=end

fn main() -> Result<()> {
    let nums: Vec<i32> = deserialize(&read_line()?)?;
    let ans: bool = Solution::contains_duplicate(nums);

    println!("\noutput: {}", serialize(ans)?);
    Ok(())
}
